applications/DG-Max/Algorithms/DGMaxEigenvalue.cpp

```diff
@@ -694,6 +806,19 @@ void DGMaxEigenvalue<DIM>::SolverWorkspace::solve(
     extractEigenVectors();
 }
 
+template <std::size_t DIM>
+void DGMaxEigenvalue<DIM>::SolverWorkspace::writeAsTimeIntegrationVector(
+    std::size_t eigenvectorId, std::size_t timeIntegrationVectorId) {
+    logger.assert_debug(eigenvectorId < convergedEigenValues_,
+                        "Eigenvalue % is more than converged %", eigenvectorId,
+                        convergedEigenValues_);
+    // Distribute the solution coefficients to the local element vectors
+    PetscErrorCode err;
+    err = VecCopy(eigenVectors_[eigenvectorId], tempFieldVector_);
```

Comment:
For (at least) two reasons:
 1. There is some overhead in constructing a distributed vector. As there needs to be MPI-communication about the local and global sizes. Depending on the amount of usage this may become quite a waste of performance.
 2. To reduce the clutter in the computations. Many of the computations need an intermediate vector to temporarily store a partial or intermediate result. Having to create a new vector for this does not only have a performance cost (point 1) but also clutters the code.

There is of course the downside that you need to look out for not reusing the same temporary vector. But usually this is quite trivial, as the called functions are more low-level and won't have access.